### PR TITLE
add method for pagination

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

File: BaseApiBundle/Repository/ApiClientRepository.php [hunk 1]
```diff
@@ -20,4 +20,134 @@ public function findOneByKeyAndSecret($key, $secret)
...
+     * @param array|null  $descriptionEntity
+     * @param array|null  $columns
+     * @param string|null $search
+     * @param array|null  $order
+     * @param int|null    $skip
+     * @param int|null    $limit
+     *
+     * @return array
+     */
+    public function findForPaginateAndSearch($descriptionEntity = null, $columns = null, $search = null, $order = null, $skip = null, $limit = null)
```
Comment (ngilain) [inline BaseApiBundle/Repository/ApiClientRepository.php:34]: This method name seems not enough explicit. Can you rename it to give it more sense ?

File: BaseApiBundle/Repository/ApiClientRepository.php [hunk 2]
```diff
@@ -20,4 +20,134 @@ public function findOneByKeyAndSecret($key, $secret)
...
+    }
+
+    /**
+     * @param array|null   $columns
+     * @param array|null   $descriptionEntity
+     * @param array|null   $search
+     *
+     * @return int
+     */
+    public function countFilterSearch($descriptionEntity = null, $columns = null, $search = null)
```
Comment (ngilain) [inline BaseApiBundle/Repository/ApiClientRepository.php]: I think there is also a problem with the naming here, some might think the method is counting the quantity of filters.

**Comment by itkg-alavieille** [comment]
@itkg-ngilain

**Comment by alavieille** [comment]
@itkg-ngilain

File: BaseApiBundle/Repository/ApiClientRepository.php [hunk 3]
```diff
@@ -20,4 +20,135 @@ public function findOneByKeyAndSecret($key, $secret)
...
+     */
+    protected function getFilterSearchField($value, $type)
+    {
+        if ($type == 'integer') {
+            $filter = (int) $value;
+        } elseif ($type == 'boolean') {
+            $value = ($value === 'true' || $value === '1') ? true : false;
+            $filter = $value;
+        } else {
+            $filter = new \MongoRegex('/.*'.$value.'.*/i');
```
Comment (veolys-nanne) [inline BaseApiBundle/Repository/ApiClientRepository.php:88]: You should use preg_quote to protect $value.

**Comment by alavieille** [comment]
@itkg-nanne

**Comment by veolys-nanne** [comment]
:+1:

**Pull request merged**